include/oneapi/tbb/detail/_concurrent_skip_list.h

```diff
@@ -700,9 +731,13 @@ class concurrent_skip_list {
     }
 
     void internal_move(concurrent_skip_list&& other) {
-        my_head_ptr.store(other.my_head_ptr.load(std::memory_order_relaxed), std::memory_order_relaxed);
+        node_ptr head = my_head_ptr.exchange(other.my_head_ptr.load(std::memory_order_relaxed), std::memory_order_relaxed);
         other.my_head_ptr.store(nullptr, std::memory_order_relaxed);
 
+        if (head != nullptr) {
+            delete_node(head);
+        }
```

Comment:
If I understand correct, the problem takes place when `internal_move` is called during move assignment. I have checked the code - we call `clear()` method before the `internal_move`. The problem happends because `clear()` does not delete the head node for the optimization.

Reply:
actually `clear()` is called in the `operator=` of the `concurrent_skip_list`. 

However this part of the patch is moved to separate PR

Reply:
moved to #409